extensions/arc/src/ui/dashboards/postgres/postgresParametersPage.ts

```diff
@@ -290,211 +290,261 @@ export class PostgresParametersPage extends DashboardPage {
-					// loading icon forever)
-					if (err instanceof UserCancelledError) {
-						vscode.window.showWarningMessage(loc.pgConnectionRequired);
-					} else {
-						vscode.window.showErrorMessage(loc.fetchEngineSettingsFailed(this._postgresModel.info.name, err));
-					}
-					this._postgresModel.engineSettingsLastUpdated = new Date();
-					this._postgresModel._onEngineSettingsUpdated.fire(this._postgresModel._engineSettings);
-					this.connectToServerButton!.enabled = true;
-					throw err;
-				});
-
+				this._parametersTableLoading!.loading = true;
+				await this.callGetEngineSettings().finally(() => this._parametersTableLoading!.loading = false);
+				this.searchBox!.enabled = true;
+				this.resetAllButton!.enabled = true;
 				this.parameterContainer!.clearItems();
 				this.parameterContainer!.addItem(this.parametersTable);
-
-			}));
+			})
+		);
 	}
 
-	private initializeSearchBox() {
+	private initializeSearchBox(): void {
 		this.searchBox = this.modelView.modelBuilder.inputBox().withProps({
 			readOnly: false,
+			enabled: false,
 			placeHolder: loc.searchToFilter
 		}).component();
 
 		this.disposables.push(
 			this.searchBox.onTextChanged(() => {
-				this.filterParameters();
+				if (!this.searchBox!.value) {
+					this.parametersTable.data = this._parameters.map(p => [p.parameterName, p.valueContainer, p.description, p.resetButton]);
+				} else {
+					this.filterParameters(this.searchBox!.value);
+				}
 			})
 		);
 	}
 
-	private filterParameters() {
-		//TODO
+	private filterParameters(search: string): void {
+		let filterData: ParametersModel[] = [];
+
+		this._parameters.forEach(param => {
+			if (param.parameterName?.search(search) !== -1 || param.description?.search(search) !== -1) {
+				filterData.push(param);
+			}
+		});
+
+		this.parametersTable.data = filterData.map(f => [f.parameterName, f.valueContainer, f.description, f.resetButton]);
 	}
 
-	private createParametersTable(): any[] {
-		/*Define server settings that shouldn't be modified. we block archive_*, restore_*, and synchronous_commit to prevent the user
-		from messing up our backups. (we rely on synchronous_commit to ensure WAL changes are written immediately.)
-		we block log_* to protect our logging. we block wal_level because Citus needs a particular wal_Level to rebalance shards
-		TODO: Review list of blacklisted parameters. wal_level should only be blacklisted if sharding is enabled
-		To not be modified
-		"archive_command", "archive_timeout", "log_directory", "log_file_mode", "log_filename", "restore_command",
-		"shared_preload_libraries", "synchronous_commit", "ssl", "unix_socket_permissions", "wal_level" */
-
-		// For ev in this._postgresModel._engineSettings
-		// create row
-		// return rows
-		this.parameterComponents('engineSetting', '');
-		return [];
+	private createParameters(): void {
+		this._parameters = [];
+		this._postgresModel._engineSettings.forEach(engineSetting => {
```

Comment:
`this._parameters = this._postgresModel._engineSettings.map(engineSetting => this.createParameterComponents(engineSetting));`